src/component/BookList.js

```diff
@@ -1,22 +1,29 @@
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
+import React, { useEffect } from 'react';
 import Book from './Book';
+import { fetchUsers } from '../redux/book/booksSlice';
 
 export default function BookList() {
   const { totalbooks } = useSelector((state) => state.book);
-  const updateid = totalbooks.map((b) => ({
-    ...b,
-  }));
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchUsers());
+  });
```

Comment:
- [ ] Well! In your current implementation, it appears that the `fetchUsers()` function is being dispatched continuously, which may not be efficient or convenient. To enhance the functionality and prevent unnecessary dispatching of `fetchUsers()`, I recommend implementing a way so that it only dispatches `fetchUsers()` when user activity is detected.

[React-App (2).webm](https://github.com/MayPyone/Bookstore/assets/55840999/c5d259d1-a332-4b19-b674-26c4a8ccdd85)

By updating your code in this manner, you'll optimize the performance and responsiveness of your application, ensuring that API calls are made when needed, rather than continuously. Keep up the excellent work! 👍🕐📚